modules/request-monitor.js

```diff
@@ -26,27 +28,85 @@ const KNOWN_FEED_TYPES = [
     'application/atom+xml',
     'application/rss+xml',
 ];
-/*
+
 // Let's not consider 'application/octet-stream' for now
-// and leave the XML-based MIME types for the content script
 const MAYBE_FEED_TYPES = [
-    'application/octet-stream',
+    //'application/octet-stream',
     'application/xml',
     'text/html',
     'text/xml',
     // and anything with `+xml` as a special case
 ];
-*/
 
-function headerCheck({tabId, url, responseHeaders}) {
+const FEED_ROOT_TAGS = ['rss', 'feed', 'rdf:RDF'];
+const SNIFF_WINDOW = 512; // Matches the legacy Firefox feed sniffer window
```

Comment:
The standard uses a window of 1445 bytes (or more), or until user-agent-specified timeout.